src/Elastic.Apm/Logging/IApmLoggingExtensions.cs

```diff
@@ -1,60 +1,99 @@
+		/// in case the log won't be printed because the loglevel would not allow it.
+		/// </summary>
+		/// <param name="logger">The logger instance you want to log with</param>
+		/// <returns>Either a MaybeLogger or null</returns>
+		internal static MaybeLogger? Warning(this IApmLogger logger) => IfLevel(logger, LogLevel.Warning);
 
-		public static void LogDebugException(this IApmLogger logger, Exception exception) =>
-			logger?.LogDebug(exception, "{ExceptionName}: {ExceptionMessage}", exception?.GetType().Name, exception?.Message);
+		/// <summary>
+		/// If the logger has a loglevel, which is higher than Info then it returns a MaybeLogger instance,
+		/// otherwise it returns null.
+		/// By using the return value with `?.` you can avoid executing code that is not necessary to execute
+		/// in case the log won't be printed because the loglevel would not allow it.
+		/// </summary>
+		/// <param name="logger">The logger instance you want to log with</param>
+		/// <returns>Either a MaybeLogger or null</returns>
+		internal static MaybeLogger? Info(this IApmLogger logger) => IfLevel(logger, LogLevel.Information);
 
-		public static void LogErrorException(this IApmLogger logger, Exception exception,
-			[System.Runtime.CompilerServices.CallerMemberName]
-			string method = "",
-			[System.Runtime.CompilerServices.CallerFilePath]
-			string filepath = "",
-			[System.Runtime.CompilerServices.CallerLineNumber]
-			int lineNumber = 0
-		)
+		internal readonly struct MaybeLogger
 		{
-			var file = string.IsNullOrEmpty(filepath) ? string.Empty : new FileInfo(filepath).Name;
-			logger?.LogError(exception, "{ExceptionName} in {Method} ({File}:{LineNumber}): {ExceptionMessage}", exception?.GetType().Name, method, file, lineNumber, exception?.Message);
-		}
+			private readonly IApmLogger _logger;
+			private readonly LogLevel _level;
 
-		public static ScopedLogger Scoped(this IApmLogger logger, string scope) =>
-			 new ScopedLogger((logger is ScopedLogger s ? s.Logger : logger), scope);
+			public MaybeLogger(IApmLogger logger, LogLevel level) => (_logger, _level) = (logger, level);
 
-		private static void DoLog(this IApmLogger logger, LogLevel level, string message, Exception e, object[] args)
-		{
-			var formatter = logger is ScopedLogger sl
-				? sl.GetOrAddFormatter(message, args.Length)
-				: Formatters.GetOrAdd(message, s => new LogValuesFormatter(s));
+			public void Log(string message, params object[] args) => _logger.DoLog(_level, message, null, args);
 
-			var logValues = formatter.GetState(args);
-			logger?.Log(level, logValues, e, (s, _) => formatter.Format(args));
+			public void LogException(Exception exception, string message, params object[] args) =>
+				_logger.DoLog(_level, $"Exception: {{ExceptionType}}, Message: {{ExceptionMsg}} \n{message}", exception,
+					exception.GetType().FullName, exception.Message, args);
+
+			public void LogExceptionWithCaller(Exception exception,
```

Comment:
I think we should always have a message explaining the meaning of log statement. For example, currently the only usage of `LogExceptionWithCaller` in `PayloadSenderV2.cs` should probably mention that `TaskCanceledException` is expected when Agent PayloadSender shuts down.

Reply:
For that we have the other method (`public void LogException(Exception exception, string message, params object[] args)`), and you are right, that's what we use most of the time. 

This one is intended when you only want to log the exception. We also use this in `HttpDiagnosticListener.OnError`, probably we should do it also in other listeners; in those cases I'm not really sure we can add a more meaningful error message than having the caller method + the exception details.

Reply:
I still think a useful message can be provided in almost any context and definitely for the ones you have mentioned.
Another reason for always having message as part of logging statement is that this way logging statement can serve two purposes for the price of one - with s message it's also a comment explaining why surrounding code does what it does. For example in `PayloadSenderV2.cs` a message explaining that TaskCanceledException is expected when Agent PayloadSender shuts down makes it clear to a reader of the code why this exception is swallowed.

Reply:
We discussed on zoom and agreed that we will keep a method to log exception without a message for just few use cases when it makes sense and if we see in the future that it's an anti-pattern we will reconsider removing it.